src/parse/asp/grammar_parse.go

```diff
@@ -144,44 +144,51 @@ func (p *parser) parseStatement() *Statement {
 	s := &Statement{}
 	tok := p.l.Peek()
 	s.Pos = tok.Pos
+
+	var endPos Position
 	switch tok.Value {
 	case "pass":
 		s.Pass = true
-		p.l.Next()
+		endPos = p.l.Next().EndPos()
```

Comment:
I wonder if you can do this generically? like either the lexer or the parser could remember the previous position and then you grab it at the end of this function?